pcdsdevices/lens.py

```diff
@@ -286,31 +341,294 @@ def _make_safe(self):
+    size           : 2.092e-04
+    size FWHM      : 2.463e-04
+    sim_x: -0.1000, sim_y: 0.2000, scale: 0.1
+
+    Check the positions of the motors like so:
+
+    >>> sim.x
+    sim_x
+    -----
+    preset: align_position_one
+    position: -0.1
+
+    >>> sim.calib_z()
+    FWHM at lens   : 5.000e-04
+    waist          : 3.113e-07
+    waist FWHM     : 3.666e-07
+    rayleigh_range : 1.965e-03
+    focal length   : 2.680e+00
+    size           : 2.092e-04
+    size FWHM      : 2.463e-04
+    80
+
+    >>> sim.beam_size()
+    FWHM at lens   : 5.000e-04
+    waist          : 3.113e-07
+    waist FWHM     : 3.666e-07
+    rayleigh_range : 1.965e-03
+    focal length   : 2.680e+00
+    size           : 2.092e-04
+    size FWHM      : 2.463e-04
+    0.00024626624937199417
+
+    Use `calib_z` and `beam_size` to to move the z motor:
+
+    >>> sim.calib_z(sim.x.position, sim.y.position, sim.z.position)
+    >>> sim.beam_size(sim.x.position, sim.y.position, sim.z.position)
+
+    Calculated the pseudo position:
+
+    >>> sim.inverse(sim.x.position, sim.y.position, sim.z.position)
+    FWHM at lens   : 5.000e-04
+    waist          : 3.113e-07
+    waist FWHM     : 3.666e-07
+    rayleigh_range : 1.965e-03
+    focal length   : 2.680e+00
+    size           : 2.092e-04
+    size FWHM      : 2.463e-04
+    SimLensStackPseudoPos(calib_z=0, beam_size=0.00024626624937199417)
+
+    Calculate the real Position:
+
+    >>> sim.forward(3, 5)
+    FWHM at lens   : 5.000e-04
+    waist          : 3.113e-07
+    waist FWHM     : 3.666e-07
+    rayleigh_range : 1.965e-03
+    focal length   : 2.680e+00
+    size           : 2.092e-04
+    size FWHM      : 2.463e-04
+    SimLensStackRealPos(x=-0.1, y=0.4, z=-5359883.247959933)
+
+    """
     def __init__(self, *args, path, **kwargs):
+
         self.path = path
-        lens_set = self.read_lens()
+        self.lens_pack = self.read_lens()
+        # Defaulting this a the first set in the file
+        # TODO: this will have to change
+        lens_set = calcs.get_lens_set(1, self.path)
+
         super().__init__(*args, lens_set=lens_set, **kwargs)
 
-    def read_lens(self):
-        with open(self.path, 'r') as f:
-            read_data = yaml.safe_load(f)
-        return read_data
+    def read_lens(self, print_only=False):
+        """
+        Read the lens sets from file provided in the path.
+
+        Parameters
+        ----------
+        print_only : bool
+            To indicate if printing out the currents sets only. If `False`
+            the `self.lens_pack` will be set to get the current file sets.
+
+        Returns
+        -------
+        sets : list
+            Pack of lens sets.
+        """
+        lens_pack = calcs.get_lens_set(None, self.path, get_all=True)
+        if print_only:
+            print(lens_pack)
+        else:
+            self.lens_pack = lens_pack
+            return self.lens_pack
+
+    # TODO: i think i'll need this one here or something similar
```

Comment:
I'm leaving this here for now because I don't know yet how we want to get the set, for now I have a method `get_lens_set(index)` to manually do it

Reply:
I think `get_lens_set(index)` is a good way to handle this. Another approach could be to combine with the XFLS class and read off the y motor state, but that's not necessarily guaranteed to be consistent (what if we're between states? e.g. for a CCM experiment, which shifts the beam along the y axis)